include/git2/hook.h

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (C) the libgit2 contributors. All rights reserved.
+ *
+ * This file is part of libgit2, distributed under the GNU GPL v2 with
+ * a Linking Exception. For full terms see the included COPYING file.
+ */
+#ifndef INCLUDE_git_hook_h__
+#define INCLUDE_git_hook_h__
+
+#include "common.h"
+#include "types.h"
+
+/**
+ * @file git2/hook.h
+ * @brief Git hook management routines
+ * @defgroup git_hook Git hook management routines
+ * @ingroup Git
+ * @{
+ */
+GIT_BEGIN_DECL
+
+/**
+ * A helper to get the path of the repository's hooks.
+ *
+ * This will obey core.hooksPath.
+ *
+ * @param out_dir A pointer to a git_buf that will be filled with the absolute path.
+ * @param repo The repository to get the hooks location from.
+ * @return 0 on success, or an error code.
+ */
+GIT_EXTERN(int) git_hook_dir(git_buf *out_dir, git_repository *repo);
+
+typedef int (*git_hook_foreach_cb)(const char *hook_name, void *payload);
+
+/**
+ * Enumerate a repository's hooks.
+ *
+ * The callback will be called with the name of each available hook.
+ *
+ * @param repo The repository.
+ * @param callback The enumeration callback.
+ * @param payload A user-provided pointer that will be passed to the callback.
+ * @return 0 on success, or an error code.
+ */
+GIT_EXTERN(int) git_hook_foreach(
+	git_repository *repo,
+	git_hook_foreach_cb callback,
+	void *payload);
+
+/**
+ * A hook environment
+ *
+ * This structure will be provided by the library when a hook needs to
+ * be executed.
+ *
+ * - `path` is the absolute path to the hook executable.
+ * - `args` is the argument list for the hook.
+ * - `io` is either an in/out pointer to either data that must be provided to
+ *        the hook, a pointer where the executor will stash data outputted by
+ *        the hook, or both.
```

Comment:
Yeah. I'm no that familiar with Doxygen, I just searched for examples in our codebase any by chance found mostly `*` as bullet points.